Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -7914,6 +7920,8 @@ void WebPage::didCommitLoad(WebFrame* frame)
     bool viewportChanged = false;
 
     setCanIgnoreViewportArgumentsToAvoidExcessiveZoomIfNeeded(m_viewportConfiguration, coreFrame.get(), shouldIgnoreMetaViewport());
+
+    setCanIgnoreViewportArgumentsToAvoidEnlargedViewIfNeeded(m_viewportConfiguration, coreFrame.get(), shouldIgnoreMetaViewport());
```

Comment:
shouldn't this one just be false?

Reply:
Hm... at navigation commit-time, I'd _expect_ the desktop website policy flag (which determines whether we opt in to `MetaViewportPolicy::Ignore`) to be set already, since that's determined at navigation decision time (or if the client sets the default webpage prefs through the configuration). I haven't tested this though..

Reply:
If I set it to false then this doesn't work. I'm not super familiar with the shouldIgnoreMetaViewport() code, so if we think this is wrong I might need additional eye.

Reply:
I am confused, then. I would expect this bit to only be driven by the quirk, and nothing else.

Reply:
Wenson is right that I had the order wrong, but I'm confused why we care about `shouldIgnoreMetaViewport` at all for this quirk, I guess.

Reply:
@hortont424 did you mean "true" instead of "false"? Cause that works. And it's what I'm doing in the other call as well.

Reply:
I misunderstood the flow a bit.

I *think* that you can just remove the boolean arg entirely, never consult 'shouldIgnoreMetaViewport', and always set it to true in setCanIgnoreViewportArgumentsToAvoidEnlargedViewIfNeeded in the case where the quirk is enabled.

(I still don't understand why the previous quirk was predicated on shouldIgnoreMetaViewport, maybe Wenson can enlighten us, but I don't think this one should be).

Reply:
The intent was to limit this quirk to only when the desktop website is requested.

If the mobile site is requested on a large screen, then I'd thought it's probably best to return to the old (zoomed-in) behavior.